test/jdk/java/awt/print/PrinterJob/XparColor.java

```diff
@@ -75,184 +108,145 @@ public static void main(String s[]) {
         }
     }
 
-    public int print(Graphics g, PageFormat pf, int pi)
-        throws PrinterException {
+    private static void pass() {
+        testPassed = true;
+        countDownLatch.countDown();
+    }
+
+    private static void fail(String failureMsg) {
+        testFailureMsg = failureMsg;
+        testPassed = false;
+        countDownLatch.countDown();
+    }
+
+    private static String convertMillisToTimeStr(int millis) {
+        if (millis < 0) {
+            return "00:00:00";
+        }
+        int hours = millis / 3600000;
+        int minutes = (millis - hours * 3600000) / 60000;
+        int seconds = (millis - hours * 3600000 - minutes * 60000) / 1000;
+        return String.format("%02d:%02d:%02d", hours, minutes, seconds);
+    }
+
+    private static void createAndShowTestDialog() {
+        String testInstruction = "This testcase will be launched twice on mac platform and once on non mac platform.\n" +
+                "On Mac once with default pipeline and once again with Metal pipeline.\n" +
```

Comment:
I think it will be better to rephrase this line with "This test will be run with default pipeline in all OS and additionally with metal pipeline only in macOS"

Reply:
although I am not sure why it is important to run in metal pipeline. I don't think metal has any effect on printing par se.